optimum/habana/accelerate/accelerator.py

```diff
@@ -366,8 +368,10 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
         if self.state.dynamo_plugin.backend != DynamoBackend.NO and not is_compiled_module(model):
             compile_kwargs = self.state.dynamo_plugin.to_kwargs()
             ############################################################################################################
-            if self.use_regional_compilation:
-                compile_regions(model, **compile_kwargs)
+            if self.dynamic is not None:
```

Comment:
why didn't you just update kwargs in the same way like you did in next case?

Reply:
I just followed the old code before #1876. Will check